scripts/release.sh

```diff
@@ -458,22 +472,24 @@ function parse_flags() {
 # Run tests (unless --skip-tests was passed). Conveniently displays a banner indicating so.
 # Parameters: $1 - executable that runs the tests.
 function run_validation_tests() {
-  if (( ! SKIP_TESTS )); then
-    banner "Running release validation tests"
-    # Run tests.
-    if ! $1; then
-      banner "Release validation tests failed, aborting"
-      exit 1
-    fi
+  (( SKIP_TESTS )) && return
+  banner "Running release validation tests"
+  # Run tests.
+  if ! $1; then
+    banner "Release validation tests failed, aborting"
+    abort "release validation tests failed"
   fi
 }
 
-# Publishes the generated artifacts to GCS, GitHub, etc.
+# Publishes the generated artifacts to directory, GCS, GitHub, etc.
 # Parameters: $1..$n - files to add to the release.
 function publish_artifacts() {
   (( ! PUBLISH_RELEASE )) && return
   tag_images_in_yamls ${ARTIFACTS_TO_PUBLISH}
-  publish_to_gcs ${ARTIFACTS_TO_PUBLISH}
+  if [[ -n "${RELEASE_DIR}" ]]; then
+    cp ${ARTIFACTS_TO_PUBLISH} ${RELEASE_DIR} || abort "cannot copy release to '${RELEASE_DIR}'"
```

Comment:
where do the files end up by default? I'd prefer if they appeared only in the destination dir so that 1) we don't run the risk of them over-writing existing files, and 2) we don't leave them around by using a `cp` instead of `mv`.  And switching this to a `mv` wouldn't be good since then the net result would be that we erased a pre-existing file with the same name.

Reply:
It's up to the caller of this script to define where the manifests are written to. Specifically for serving, they're written to a temporary directory (https://github.com/knative/serving/blob/bcc67e3b34b2e6d1f9e567d1fb53aeddc4d173db/hack/generate-yamls.sh#L46). So in the end, at least for serving, we're achieving the goal you desire.